### PR TITLE
Fix Windows CI

### PR DESCRIPTION
**Pull request by alexeagle**

### CODE REVIEW

File: .ci/rules_nodejs.json [hunk 1]
```diff
@@ -16,5 +16,4 @@
       ]
     }
   }
-
-]
\ No newline at end of file
+]
```
Comment (alexeagle) [inline .ci/rules_nodejs.json]: revert

File: examples/rollup/BUILD.bazel [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # NOTE: the example uses //:defs.bzl because it lives in the same WORKSPACE
 # as the rule declarations.
 # Users should use @build_bazel_rules_nodejs//:defs.bzl
```
Comment (alexeagle) [inline examples/rollup/BUILD.bazel]: update comment

File: internal/jasmine_runner.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 const fs = require('fs');
 const path = require('path');
-const JasmineRunner = require('jasmine');
+const JasmineRunner = require('jasmine/lib/jasmine.js');
```
Comment (alexeagle) [inline internal/jasmine_runner.js]: doesn't need .js

File: internal/jasmine_node_test.bzl [hunk 4]
```diff
@@ -7,11 +7,9 @@ def jasmine_node_test(name, srcs, data = [], args = [], deps = [], **kwargs):
       deps = srcs + deps,
       testonly = 1,
   )
+  manifest = Label("//%s:%s_devmode_srcs.MF" % (PACKAGE_NAME, name))
 
-  args = ["/".join([p
-      for p in [".", PACKAGE_NAME, "%s_devmode_srcs.MF" % name]
-      if p
-  ])] + args
+  args = ["/".join([manifest.workspace_root.split("/")[1], manifest.package, manifest.name])] + args
```
Comment (alexeagle) [inline internal/jasmine_node_test.bzl]: check if we still need this change

File: internal/node.bzl [hunk 5]
```diff
@@ -78,17 +78,23 @@ def _nodejs_binary_impl(ctx):
       ])
     ctx.template_action(
         template=ctx.file._launcher_template,
-        output=ctx.outputs.executable,
+        output=ctx.outputs.main,
         substitutions={
             "TEMPLATED_node": ctx.workspace_name + "/" + node.path,
-            "TEMPLATED_args": " ".join(ctx.attr.args),
+            "TEMPLATED_args": "",
```
Comment (alexeagle) [inline internal/node.bzl]: need a todo here? restore the args?

File: internal/node.bzl [hunk 6]
```diff
@@ -117,20 +123,38 @@ _NODEJS_EXECUTABLE_ATTRS = {
 }
 
 _NODEJS_EXECUTABLE_OUTPUTS = {
-    "loader": "%{name}_loader.js"
+    "loader": "%{name}_loader.js",
+    "main": "%{name}_launcher.sh"
 }
 
-nodejs_binary = rule(
+nodejs_binary_rule = rule(
```
Comment (alexeagle) [inline internal/node.bzl]: users will see this name in `bazel query --output=label_kind` so we should make this name match the name they use

File: internal/node.bzl [hunk 7]
```diff
@@ -117,20 +123,38 @@ _NODEJS_EXECUTABLE_ATTRS = {
...
+    )
+
+    native.sh_binary(
+        name = name,
+        args = args,
+        srcs = [":%s_loader_launcher.sh" % name],
+        data = [":%s_loader" % name],
+    )
+
+def nodejs_test(name, args=[], **kwargs):
```
Comment (alexeagle) [inline internal/node.bzl]: comment how it's duplicate of above intentionally

File: internal/node_launcher.sh [hunk 8]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/bin/bash
```
Comment (alexeagle) [inline internal/node_launcher.sh]: is this change needed?

File: internal/node_launcher.sh [hunk 9]
```diff
@@ -44,7 +44,10 @@ case "$0" in
  *) self="$PWD/$0" ;;
 esac
 
-if [[ -n "$TEST_SRCDIR" ]]; then
+if [[ -n "$RUNFILES_MANIFEST_ONLY" ]]; then
+  # Windows only has a manifest file instead of symlinks.
+  RUNFILES=${RUNFILES_MANIFEST_FILE%/MANIFEST}
```
Comment (alexeagle) [inline internal/node_launcher.sh]: cleanup: we trim the `/MANIFEST` only to add it back later

File: internal/node_loader.js [hunk 10]
```diff
@@ -102,6 +101,7 @@ module.constructor._resolveFilename =
   var resolveLocations = [
     request,
     resolveRunfiles(request),
+    resolveRunfiles('TEMPLATED_workspace_name', 'TEMPLATED_label_package', request),
```
Comment (alexeagle) [inline internal/node_loader.js]: shouldn't need to add a fourth lookup location, at least without understanding why

Comment (filipesilva) [reply internal/node_loader.js]: I needed the fourth lookup to be able to import `bundle.js` in the rollup jasmine tests.

File: internal/node_launcher.sh (hunk 8 above)
Comment (filipesilva) [reply internal/node_launcher.sh]: At some point I needed this change to run on windows, but we should double check if it's still needed.

Comment (alexeagle) [reply internal/node_launcher.sh]: confirmed, it's still needed.

File: .ci/rules_nodejs.json [hunk 11]
```diff
@@ -11,9 +11,11 @@
...
         "-slow"
       ],
+      "targets": ["//...", "@program_example/..."],
       "configure": [
         "%BAZEL% run @yarn//:yarn"
       ]
     }
   }
-]
+
```
Comment (filipesilva) [inline .ci/rules_nodejs.json]: Nit: remove empty line.

File: internal/node_loader.js [hunk 12]
```diff
@@ -107,7 +107,6 @@ module.constructor._resolveFilename =
   var resolveLocations = [
     request,
     resolveRunfiles(request),
-    resolveRunfiles('TEMPLATED_workspace_name', 'TEMPLATED_label_package', request),
```
Comment (filipesilva) [inline internal/node_loader.js]: I take it that removing this related to https://github.com/bazelbuild/rules_nodejs/issues/32?

File: internal/jasmine_node_test.bzl [hunk 13]
```diff
@@ -1,38 +1,23 @@
...
-  args = ["/".join([p
-      for p in [".", "external" if is_external else "",
-                target_repository, PACKAGE_NAME, "%s_devmode_srcs.MF" % name]
-      if p
-  ])] + args
-
   data += srcs + deps
   data += [Label("//internal:jasmine_runner.js")]
-  data += [manifest]
+  data += [":%s_devmode_srcs.MF" % name]
```
Comment (filipesilva) [inline internal/jasmine_node_test.bzl]: 👍

**Comment by alexeagle** [comment]
FYI @dslomov :)

**Comment by dslomov** [comment]
Cool!
